orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v1schema/V1SchemaExecutionGenerator.java

```diff
@@ -0,0 +1,85 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.orca.pipelinetemplate.v1schema;
+
+import com.netflix.spinnaker.orca.pipelinetemplate.generator.ExecutionGenerator;
+import com.netflix.spinnaker.orca.pipelinetemplate.v1schema.model.PipelineTemplate;
+import com.netflix.spinnaker.orca.pipelinetemplate.v1schema.model.PipelineTemplate.Configuration;
+import com.netflix.spinnaker.orca.pipelinetemplate.v1schema.model.StageDefinition;
+import com.netflix.spinnaker.orca.pipelinetemplate.v1schema.model.TemplateConfiguration;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+public class V1SchemaExecutionGenerator implements ExecutionGenerator {
+
+  @Override
+  public Map<String, Object> generate(PipelineTemplate template, TemplateConfiguration configuration) {
+    Map<String, Object> pipeline = new HashMap<>();
+    pipeline.put("id", configuration.getRuntimeId());
+    pipeline.put("application", configuration.getPipeline().getApplication());
+    pipeline.put("name", Optional.ofNullable(configuration.getPipeline().getName()).orElse("Unnamed Execution"));
+    // TODO Doesn't look like we even use this anywhere in the codebase.
+//    pipeline.put("appConfig", "unsupported");
+
+    // TODO rz - Ehhhh
+    Configuration c = template.getConfiguration();
+    if (template.getConfiguration() == null) {
+      pipeline.put("parallel", true);
+      pipeline.put("limitConcurrent", true);
+      pipeline.put("keepWaitingPipelines", false);
+      pipeline.put("notifications", Collections.emptyList());
+    } else {
+      pipeline.put("parallel", c.getConcurrentExecutions().getOrDefault("parallel", true));
+      pipeline.put("limitConcurrent", c.getConcurrentExecutions().getOrDefault("limitConcurrent", true));
+      pipeline.put("keepWaitingPipelines", c.getConcurrentExecutions().getOrDefault("keepWaitingPipelines", false));
+      pipeline.put("notifications", c.getNotifications());
+    }
+
+    pipeline.put("stages", template.getStages()
+      .stream()
+      .map(s -> {
+        Map<String, Object> stage = new HashMap<>();
+        stage.put("id", UUID.randomUUID().toString());
+        stage.put("refId", s.getRefId());
+        stage.put("type", s.getType());
+        stage.put("name", s.getName());
+        stage.put("context", s.getConfig());
+        stage.put("requisiteStageRefIds", getStageRequisiteIds(s, template.getStages()));
+        return stage;
+      })
+      .collect(Collectors.toList()));
+
+    return pipeline;
+  }
+
+  private static List<Integer> getStageRequisiteIds(StageDefinition stage, List<StageDefinition> allStages) {
```

Comment:
I've moved this into the graph transforms in a PR I'll be opening after this is merged.